subprojects/packagefiles/fio/meson.build

```
@@ -0,0 +1,37 @@
+project(
+  'fio',
+  'c',
+  version: '3.28',
+)
+fs = import('fs')
+
+if not fs.exists('config-host.h')
+  message('Configuring ..')
+  run_command('./configure', capture: false, check: true)
+endif
+if not fs.exists('fio')
+  cpu_count = run_command([
+    import('python').find_installation('python3'),
+    '-c',
+    'from multiprocessing import cpu_count; print(cpu_count())'
+    ],
+    check: true,
+  ).stdout().strip()
+
+  message('Building ..')
+  run_command([
+      find_program(host_machine.system() == 'freebsd' ? 'gmake' : 'make'),
```

```suggestion
      find_program('gmake', 'make'),
```

will check for both, in turn, and a gmake binary is never incorrect, even on systems other than FreeBSD. Although it would find BSD make if gmake is not installed at all... you could solve that by requiring a specific minimum version of make, which satisfies `gmake --version` but fails on BSD make since the latter doesn't have a `--version` flag.

Quite frankly, the entire approach here is totally wrong IMO. Why are you building this subproject using a Makefile instead of porting it to a real meson.build? How are you installing it?

If you must use the external build system as is, consider using https://mesonbuild.com/External-Project-module.html

---

This is a very good point. I don't think we are doing anything fancy with the installation. AFAIK, fio is just a `make ; make install`. @jwdevantier, Any inconvenience in using the meson way?

Additionally, we don't really make an effort to build on freeBSD, not test for it. At this point, I would not worry about the freebsd caes.

---

**1) How to determine whether to use `make` or `gmake`**
I actually the existing approach is better - it's more explicit and, as you mention, we won't end up invoking BSD make on FreeBSD by mistake which may (or may not) get us into trouble.
I think this is more explicit, if on FreeBSD, require `gmake`, otherwise error out. Having a subsequently check whether `make --version` succeeds or not seems indirect and hack'ish to me. By contrast, the existing one-line makes it obvious what is intended.

But it's interesting (and nice) that `find_program` can take multiple arguments and that it will look for them in turn, that can become useful in other situations.